File: hello.c
#include <stdio.h>
#include "datamatrix.h"

int main(int argc, char **argv)
{
	DataMatrixEncoder *encoder;
	int columns;
	int rows;
	int row;
	encoder = DataMatrixEncoder_New();
	DataMatrixEncoder_Encode(encoder, "Wikipedia");
	columns = DataMatrixEncoder_GetColumns(encoder);
	rows = DataMatrixEncoder_GetRows(encoder);
	for (row = 0; row < rows; row++) {
		int column;
		for (column = 0; column < columns; column++) {
			int module = DataMatrixEncoder_GetModule(encoder, column, row);
			putchar(module ? '#' : ' ');
		}
		putchar('\n');
	}
	return 0;
}

File: hello.pl
use DataMatrix;

my $encoder = DataMatrixEncoder->new();
$encoder->encode('Wikipedia');
for my $row (0 .. $encoder->get_rows() - 1) {
	for my $column (0 .. $encoder->get_columns() - 1) {
		print $encoder->get_module($column, $row) ? '#' : ' ';
	}
	print "\n";
}
